x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/serializer.ts

```diff
@@ -4,40 +4,133 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { SerializedPolicy } from '../../../../common/types';
+import { isEmpty } from 'lodash';
 
-import { FormInternal } from './types';
+import { AllocateAction, SerializedPhase, SerializedPolicy } from '../../../../common/types';
+
+import { FormInternal, DataAllocationMetaFields } from './types';
+import { isNumber } from '../../services/policies/policy_serialization';
+
+const unsafeSerializePhaseWithAllocation = (
```

Comment:
Do you call it unsafe because we mutate the objects? Could this method be converted to a pure function and return new objects? It seems that it would make the code easier to understand.

Reply:
Will refactor to make it a pure function instead!